common/lib/xmodule/xmodule/js/spec/video/video_caption_spec.js

```diff
@@ -158,14 +164,103 @@
                 });
             });
 
+            describe('renderCaptions', function() {
+
+                var KEY = $.ui.keyCode;
+
+                function keyPressEvent(key) {
+                    return $.Event('keydown', { keyCode: key });
+                }
+
+                describe('is rendered', function() {
+
+                    it('toggle the captions on control click', function() {
+                        state = jasmine.initializePlayer();
+
+                        $('.toggle-captions').click();
+                        expect($('.toggle-captions')).toHaveClass('is-active');
+                        expect($('.closed-captions')).toHaveClass('is-visible');
+
+                        $('.toggle-captions').click();
+                        expect($('.toggle-captions')).not.toHaveClass('is-active');
+                        expect($('.closed-captions')).not.toHaveClass('is-visible');
+                    });
+
+                    it('toggles the captions on keypress ENTER', function() {
+                        // what i originally had
+                        // state = jasmine.initializePlayer();
+
+                        // $('.toggle-captions').focus().trigger(keyPressEvent(KEY.ENTER));
+                        // expect($('.toggle-captions')).toHaveClass('is-active');
+                        // expect($('.closed-captions')).toHaveClass('is-visible');
+
+                        // $('.toggle-captions').focus().trigger(keyPressEvent(KEY.ENTER));
+                        // expect($('.toggle-captions')).not.toHaveClass('is-active');
+                        // expect($('.closed-captions')).not.toHaveClass('is-visible');
+
+                        // using examples from the platform, should work?
+                        // runs(function() {
+                        //     state = jasmine.initializePlayer();
+
+                        //     $('.toggle-captions').focus();
+                        //     $('.toggle-captions').trigger(keyPressEvent(KEY.ENTER));
+                        // });
+
+                        // waitsFor(function() {
+                        //     return $('.toggle-captions').hasClass('is-active');
+                        // }, "closed captions to be enabled", 5000);
+
+                        // runs(function() {
+                        //     expect($('.toggle-captions')).toHaveClass('is-active');
+                        //     expect($('.closed-captions')).toHaveClass('is-visible');
+                        // });
+
+                        // trying setTimeout, which does work, but should it?
```

Comment:
The [waitsFor](http://jasmine.github.io/1.3/introduction.html#section-32) method would be the way to go.   setTimeout would pass no matter what is in it.  You can even explicitly throw an error in the timeout function and the test will still pass.